### PR TITLE
Clarify auto upgrades docs

### PR DESCRIPTION
**Pull request by bernardjkim**
Addresses https://github.com/gravitational/teleport/issues/29040

- Redirect users to reinstall teleport using the system package manager before enrolling in auto-upgrades.
- Fix global scope in `enroll-agent-into-automatic-updates` page.
- Changed cloud downloads to install `teleport-ent-updater` instead of `teleport-ent`.
- Removed manual installation methods.

### CODE REVIEW

File: docs/pages/management/operations/enroll-agent-into-automatic-updates.mdx [hunk 1]
```diff
@@ -117,6 +119,16 @@ $ echo <Var name="version-server-url/path" />/<Var name="release-channel" /> | s
 Make sure not to include `https://` as a prefix to the server address.
 </Admonition>
 
+If the agent was not installed using the system package manager, follow the instructions on the
```
Comment (r0mant) [inline docs/pages/management/operations/enroll-agent-into-automatic-updates.mdx:122]: Can we just embed the same instructions here instead of redirecting users to another page? Move it into a partial if needed and include in both places.

File: docs/pages/management/operations/enroll-agent-into-automatic-updates.mdx [hunk 2]
```diff
@@ -201,6 +217,16 @@ $ sudo touch /etc/teleport-upgrade.d/schedule
 $ sudo chown <your-teleport-user> /etc/teleport-upgrade.d/schedule
 ```
 
+If the agent was not installed using the system package manager, follow the instructions on the
```
Comment (r0mant) [inline docs/pages/management/operations/enroll-agent-into-automatic-updates.mdx:220]: Same suggestion as above.

File: docs/pages/management/operations/enroll-agent-into-automatic-updates.mdx [hunk 3]
```diff
@@ -117,6 +119,16 @@ $ echo <Var name="version-server-url/path" />/<Var name="release-channel" /> | s
 Make sure not to include `https://` as a prefix to the server address.
 </Admonition>
 
+If the agent was not installed using the system package manager, follow the instructions on the
+[Teleport Enterprise Cloud Downloads page](../../choose-an-edition/teleport-cloud/downloads.mdx)
+to add the teleport package repository and reinstall teleport. To check if teleport
```
Comment (r0mant) [inline docs/pages/management/operations/enroll-agent-into-automatic-updates.mdx:124]: TBH I find these instructions pretty hard to follow/understand right now because it's just one wall of text. I think we should refactor this guide a bit to follow our normal guide structure with steps e.g.

```
## Step 1. Install Teleport updater package
// explain that Teleport needs to be installed from apt/rpm repo and how to check

## Step 2. Create updater configuration
//

## Step 3. Check for updates
//
```

No new content is actually needed, just structuring what we have a bit better. What do you think?

**Comment by bernardjkim** [comment]
@r0mant How does https://github.com/gravitational/teleport/pull/29211/commits/c4174deaf7f5fe36b2c86a80b25ac0dd852326eb look?

File: docs/pages/management/operations/enroll-agent-into-automatic-updates.mdx [hunk 4]
```diff
@@ -81,18 +81,27 @@ Server ID                            Hostname        Services Version Upgrader
...
 ```code
-$ teleport version
-Teleport Enterprise v(=teleport.version=) go(=teleport.golang=)
+$ sudo apt list teleport-ent
+# or
+$ sudo yum list teleport-ent
 ```
 
-Create the upgrade configuration directory:
+If the `teleport-ent` package is not found, add the teleport repository and reinstall teleport:
```
Comment (r0mant) [inline docs/pages/management/operations/enroll-agent-into-automatic-updates.mdx:98]:
```suggestion
If the `teleport-ent` package is not found, add the appropriate repository and reinstall Teleport:
```

File: docs/pages/management/operations/enroll-agent-into-automatic-updates.mdx [hunk 5]
```diff
@@ -81,18 +81,27 @@ Server ID                            Hostname        Services Version Upgrader
...
+<TabItem label="systemd" scope="enterprise">
 
-Confirm you have the Teleport Enterprise edition installed.
+### Step 1/3. Ensure the Teleport repository is added and Teleport Enterprise is installed
 
+To check if teleport was installed using the package manager:
 ```code
-$ teleport version
-Teleport Enterprise v(=teleport.version=) go(=teleport.golang=)
+$ sudo apt list teleport-ent
```
Comment (r0mant) [inline docs/pages/management/operations/enroll-agent-into-automatic-updates.mdx:93]: Will this show anything if the .deb or .rpm package was installed directly from a file (`dpkg -i`/`rpm -i`) rather than from a repository? Cause in this case they'd also need to add the repo source.

**Review by r0mant — APPROVED** [review]
looks good with a couple of suggestions

File: docs/pages/management/operations/enroll-agent-into-automatic-updates.mdx (hunk 5 above)
Comment (bernardjkim) [reply docs/pages/management/operations/enroll-agent-into-automatic-updates.mdx]: Good point. How about using `apt-cache policy` or `yum repolist`?

Comment (r0mant) [reply docs/pages/management/operations/enroll-agent-into-automatic-updates.mdx]: @fheinecke What would be a good way to reliably tell if a teleport package was installed from a repository or a file?

Comment (fheinecke) [reply docs/pages/management/operations/enroll-agent-into-automatic-updates.mdx]: @r0mant That depends - do you want to know if _any_ Teleport package has been installed on the system via a repository? Or do you want to know if the `tsh` command corresponds to a Teleport package installed on the system via a repository? Or something else?

As far as I am aware there is no 100% perfect way to do either, but there are a few ways to do each that are probably (hopefully) good enough for this application.

Comment (r0mant) [reply docs/pages/management/operations/enroll-agent-into-automatic-updates.mdx]: The former - to let users know whether they need to add our repository to install the updater.

Comment (bernardjkim) [reply docs/pages/management/operations/enroll-agent-into-automatic-updates.mdx]: What about just checking if `/etc/apt/sources.list.d/teleport.list` or `/etc/yum.repos.d/teleport.repo` exists?

Comment (r0mant) [reply docs/pages/management/operations/enroll-agent-into-automatic-updates.mdx]: @bernardjkim Sounds good to me, this is what our install instructions recommend.

**Review by lsgunn-teleport — APPROVED** [review]
LGTM

**Comment by alexfornuto** [comment]
Removed from queue because I see edits required, will review now.

File: docs/pages/includes/cloud/install-linux-cloud.mdx [hunk 6]
```diff
@@ -15,7 +15,7 @@
   | sudo tee /etc/apt/sources.list.d/teleport.list > /dev/null
 
   $ sudo apt-get update
-  $ sudo apt-get install teleport-ent
+  $ sudo apt-get install teleport-ent-updater
```
Comment (alexfornuto) [inline docs/pages/includes/cloud/install-linux-cloud.mdx:18]: Is this package name replacing `teleport-ent` as the way to get the binary, or is it a separate package to provide the updater?

[the remark this answers is not in the document]
Comment (bernardjkim) [reply docs/pages/includes/cloud/install-linux-cloud.mdx:18]: > Is this package name replacing teleport-ent as the way to get the binary

I believe that was the intention, but I just noticed that install `teleport-ent-updater` doesn't install `teleport-ent` as a dependency, but it installs oss `teleport`. 

```
# yum install teleport-ent-updater
DigitalOcean Droplet Agent                                                                                                             2.8 kB/s | 3.3 kB     00:01
Dependencies resolved.
=======================================================================================================================================================================
 Package                                          Architecture                       Version                                Repository                            Size
=======================================================================================================================================================================
Installing:
 teleport-ent-updater                             x86_64                             13.2.3-1                               teleport                              13 k
Installing dependencies:
 teleport                                         x86_64                             13.2.3-1                               teleport                             131 M


```
